pkg/search/proxy/store/util.go

```diff
@@ -195,9 +195,21 @@ func (w *watchMux) AddSource(watcher watch.Interface, decorator func(watch.Event
 
 // Start run the watcher
 func (w *watchMux) Start() {
+	wg := sync.WaitGroup{}
 	for _, source := range w.sources {
-		go w.startWatchSource(source.watcher, source.decorator)
+		source := source
```

Comment:
> Reports declarations of variables that overlap with the declarations in the outer scope.

A warning from Goland IDE.

If you want to `pin` the variable, we can give it another name, or:
```diff
diff --git a/pkg/search/proxy/store/util.go b/pkg/search/proxy/store/util.go
index 00f93c467..dbb765959 100644
--- a/pkg/search/proxy/store/util.go
+++ b/pkg/search/proxy/store/util.go
@@ -196,8 +196,8 @@ func (w *watchMux) AddSource(watcher watch.Interface, decorator func(watch.Event
 // Start run the watcher
 func (w *watchMux) Start() {
        wg := sync.WaitGroup{}
-       for _, source := range w.sources {
-               source := source
+       for i := range w.sources {
+               source := w.sources[i]
                wg.Add(1)
                go func() {
                        defer wg.Done()


```

Reply:
@RainbowMango Fixed.